### PR TITLE
Surpress macOS AR warnings differently

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Use a wrapper script to remove warnings instead of injecting dummy symbols to every objects
## Impact
macOS sim
## Testing
I built sim:spiffs config locally

### CODE REVIEW

File: boards/sim/sim/sim/scripts/Make.defs [hunk 1]
```diff
@@ -69,10 +64,11 @@ CPP = $(CROSSDEV)cc -E -P -x c
 LD = $(CROSSDEV)ld
 ifeq ($(CONFIG_HOST_MACOS),y)
 STRIP = $(CROSSDEV)strip
+AR = $(TOPDIR)/tools/xcode-silent-ar-rcs
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/scripts/Make.defs:67]: Should we give the new file .sh suffix?
Change to macar like zdsar?

Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs:67]: how about macar-rcs.sh then?

Comment (xiaoxiang781216) [reply boards/sim/sim/sim/scripts/Make.defs:67]: Fine with me

Comment (yamt) [reply boards/sim/sim/sim/scripts/Make.defs]: done

**Pull request merged**